### PR TITLE
DEVDOCS-4622: [new] X-Correlation-Id header for headless storefront requests

### PR DESCRIPTION
**Pull request by slsriehl**
# [DEVDOCS-4622]

## What changed?
Provide a bulleted list in the present tense
* add ID to table of request headers, add descriptions to best practices and headless overview

## Anything else?
Add related PRs, salient notes, ticket numbers, etc.

ping @bc-cbrisky

[DEVDOCS-4692]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4692?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DEVDOCS-4622]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4622?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bc-traciporter — DISMISSED** [review]
Nice work @slsriehl. I like the GraphQL example. 
I had a couple of editorial suggestions, you can take or leave them.

File: docs/api-docs/headless/overview.mdx [hunk 1]
```diff
@@ -27,11 +27,110 @@ In the diagram below, the storefront is where the shopper interacts with product
...
-|**BigCommerce**|BigCommerce will respond to the application with the requested data to power the backend ecommerce functionality. It can handle processing payments, storing customer data, retrieving the catalog and order information.|
+| Entity | Description |
+|:-------|:------------|
+| `Storefront` | The frontend presentation layer where a shopper interacts with products. In a headless architecture, the storefront might be a CMS, native mobile app, kiosk, static site, or any other frontend solution you can imagine. The BigCommerce for WordPress plugin is built by using an existing CMS and injecting a store's catalog. You can use any CMS that accepts custom integrations. Another option is to build a storefront from scratch using a framework such as [Gatsby](https://www.bigcommerce.com/blog/flexible-headless-commerce-solutions/#overview-of-bigcommerce-for-react-gatsby). |
```
[the remark this answers is not in the document]
Comment (slsriehl) [reply docs/api-docs/headless/overview.mdx:32]: The first sentence describes the `Storefront` entity, not frontend presentation layers generally.

File: docs/api-docs/getting-started/best-practices.mdx [hunk 2]
```diff
@@ -156,6 +156,10 @@ As an alternative, consider using a server-side application to subscribe to [car
 
 BigCommerce does have limits on the number of products, categories, brands, etc. that can be created in a store. See [Platform Limits](https://support.bigcommerce.com/s/article/Platform-Limits) for more details.
 
+## Correlating requests
+
+When you are building a headless storefront, it's best practice to group related requests using the `X-Correlation-Id` header. This indicates to BigCommerce's infrastructure that an API call is part of a larger operation, and helps us track the handoff from request to request as the operation moves through our servers. To learn more about the header, see our [list of request headers](/api-docs/getting-started/about-our-api#request-headers). For an example and a more in-depth explanation, see the [Introduction to Headless Commerce](/api-docs/storefronts/guide/overview#correlating-requests).
```
Comment (slsriehl) [inline docs/api-docs/getting-started/best-practices.mdx:161]:
```suggestion
When building a headless storefront, it's best practice to group related requests using the `X-Correlation-Id` header. This indicates to BigCommerce's infrastructure that an API call is part of a larger operation, and helps us track the handoff from request to request as the operation moves through our servers. To learn more about the header, see our [list of request headers](/api-docs/getting-started/about-our-api#request-headers). For an example and a more in-depth explanation, see the [Introduction to Headless Commerce](/api-docs/storefronts/guide/overview#correlating-requests). 
```

[the remark this answers is not in the document]
Comment (slsriehl) [reply docs/api-docs/getting-started/best-practices.mdx:161]: i'm going to split the difference here.

File: docs/api-docs/headless/overview.mdx [hunk 3]
```diff
@@ -27,11 +27,110 @@ In the diagram below, the storefront is where the shopper interacts with product
...
+
+When you're using the BigCommerce for WordPress plugin, there is no need to send the `X-Correlation-Id` header.
+
+The following example uses the GraphQL Storefront API to complete a checkout and generate a payment access token, then uses that payment access token to make the payment using the Payments API.
+
+<Tabs items={['Request', 'Response']}>
+  <Tab>
+
+    ```http {6-7} title="Example request, GraphQL Storefront" showLineNumbers copy
+      POST https://example.com/graphql
```
Comment (slsriehl) [inline docs/api-docs/headless/overview.mdx:48]:
```suggestion
      POST https://store.example.com/graphql
```

**Pull request merged**